reader.lua

```diff
@@ -201,10 +201,28 @@ SettingsMigration:migrateSettings(G_reader_settings)
 
 local exit_code
 
+local function getPathFromURI(str)
+    local function decodeChar(hex)
+        return string.char(tonumber(hex,16))
+    end
+
+    local function decodeString(str)
+        local output, t = string.gsub(str,"%%(%x%x)",decodeChar)
```

Comment:
We can just leave it out if unused.
```suggestion
        local output = string.gsub(str,"%%(%x%x)",decodeChar)
```